src/libraries/pkg/baseline/packageIndex.json

```diff
@@ -543,12 +543,13 @@
         "3.1.0",
         "3.1.1",
         "3.1.2",
-        "3.1.3"
+        "3.1.3",
+        "3.1.4"
       ],
       "BaselineVersion": "5.0.0",
       "InboxOn": {},
       "AssemblyVersionInPackageVersion": {
-        "3.1.3.0": "5.0.0",
+        "3.1.4.0": "5.0.0",
```

Comment:
I’m not sure about this one but I understand you didn’t do this manually so I’m ok with it

Reply:
I **did** do this manually. 😄 

The assembly version in DependencyModel got updated to `3.1.4.0`, so I updated this as well.

Reply:
How do I do it automatically?

Reply:
You should not update this.  This is meant to indicate the first package version an assembly version appeared in.

Reply:
I believe that the break in the build error should tell you the command you have to run in order to fix this automatically

Reply:
From the error you copied on the PR description I can see that it is telling you the instructions on how to fix it

Reply:
> You should not update this. This is meant to indicate the first package version an assembly version appeared in.

I'm wondering why this line is even necessary. Can it be removed?

Reply:
Note, if I don't update this line, I get:

```
C:\Users\eerhardt\.nuget\packages\microsoft.dotnet.build.tasks.packaging\5.0.0-beta.20256.5\build\Packaging.targets(1142,5): error : PackageIndex from F:\git\runtime\src\libraries\pkg\baseline\packageIndex.json is missing an assembly version entry for 3.1.4.0 for package Microsoft.Extensions.DependencyModel.  Please run /t:UpdatePackageIndex on this project to commit an update. [F:\git\runtime\src\libraries\Microsoft.Extensions.DependencyModel\pkg\Microsoft.Extensions.DependencyModel.pkgproj]
C:\Users\eerhardt\.nuget\packages\microsoft.dotnet.build.tasks.packaging\5.0.0-beta.20256.5\build\Packaging.targets(1142,5): error : PackageIndex from F:\git\runtime\src\libraries\pkg\baseline\packageIndex.json is has an assembly version entry(s) for 3.1.3.0 which are no longer in package Microsoft.Extensions.DependencyModel.  Please run /t:UpdatePackageIndex on this project to commit an update. [F:\git\runtime\src\libraries\Microsoft.Extensions.DependencyModel\pkg\Microsoft.Extensions.DependencyModel.pkgproj]
```

Reply:
We actually do use it to determine which package dependency version other packages will get if they depend on DependencyModel. For example: say you have a library B which depends on DependencyModel, and more specifically they depend on assembly version 3.1.4.0. Our package infrastructure will then come to this table, and check which package version is the one where that assembly version first appeared, (the table above actually looks incorrect as I would assume that it should have something like `"3.1.4.0": "3.1.4"`) and then determine that package B should depend on at least DependencyModel package version 3.1.4. Assuming there is no BaselineVersion for DependencyModel (which in this case we do have one, but let's pretend we don't) then B would depend on 3.1.4 DependencyModel. In this case we do have a BaselineVersion, so even if our package infrastructure chooses package version 3.1.4 for the dependency, it will then baseline up to 5.0.0 because BaselineVersion > InitialChoosenPackageVersion.

Does that sort of make sense?

Reply:
FWIW, I believe that the right contents of that dictionary for DependencyModel should look something like:

```json
      "AssemblyVersionInPackageVersion": {
        "3.1.3.0": "3.1.3",
        "3.1.4.0": "3.1.4",
        "5.0.0.0": "5.0.0"
      }
```

Reply:
Since we are so far in the CI now, I want to wait for it to finish and merge this to unblock the build. I can put up another change for the above.

Reply:
I'm fine with that as long as we take care of it 😄

Reply:
https://github.com/dotnet/runtime/pull/36301